pkg/blobinfocache/boltdb/boltdb.go

```diff
@@ -282,13 +282,14 @@ func (bdc *cache) RecordKnownLocation(transport types.ImageTransport, scope type
 	}) // FIXME? Log error (but throttle the log volume on repeated accesses)?
 }
 
-// appendReplacementCandidates creates prioritize.CandidateWithTime values for digest in scopeBucket with corresponding compression info from compressionBucket (if compressionBucket is not nil), and returns the result of appending them to candidates.
-func (bdc *cache) appendReplacementCandidates(candidates []prioritize.CandidateWithTime, scopeBucket, compressionBucket *bolt.Bucket, digest digest.Digest, requireCompressionInfo bool) []prioritize.CandidateWithTime {
+// appendReplacementCandidates creates prioritize.CandidateWithTime values for digest in scopeBucket
+// (or with an unknown location if scope is nil and v2Output is set) with corresponding compression
+// info from compressionBucket (if compressionBucket is not nil), and returns the result of appending them
+// to candidates.
+// v2Output ensures that candidates with unknown location are also appended to the final result ignoring the
+// the field LastSeen.
```

Comment:
As is this talks about scope twice, and does not document compression.

Something like
```suggestion
// appendReplacementCandidates creates prioritize.CandidateWithTime values for digest in scopeBucket
// (which might be nil) with corresponding compression
// info from compressionBucket (which might be nil), and returns the result of appending them
// to candidates.
// v2Output allows including candidates with unknown location, and filters out candidates
// with unknown compression.
```
with some reformatting?

Same for the memory backend.